Linking.md

```diff
@@ -92,7 +91,8 @@ present:
 | Field  | Type             | Description                              |
 | ------ | ---------------- | ---------------------------------------- |
 | offset | `varuint32`      | offset of the value to rewrite           |
-| index  | `varuint32`      | the index of the function used           |
+| index  | `varuint32`      | the index of the function used, in the   |
```

Comment:
And describe it as "index into the symbol table", then we don't need to repeat it.  Maybe we should call it `symbol`?

All relocations except for TYPE_INDEX will have the same entry here so maybe we can express that so we don't need to repeat ourselves.

Reply:
I'm still imagining that these will be typed indexes, ie there'll be three vectors, one each for FUNCTION/DATA/GLOBAL symbols, and the FUNCTION_INDEX relocs will index into the FUNCTION symbols, the MEMORY relocs will index into the DATA symbols, etc...

Do you really prefer the single index space? We could do that, it's just a lot of type checks every time you use an index you have to check or assert "is the symbol at this index of the expected type". There might be a way to do all those checks up-front though. Maybe coding up the prototype would make it clearer which way is shorter.

Reply:
Yes, a single symbol lndex space makes the most sense to me.    This is in line with how woth object formats work I think, and it means we can add new symbol types without adding new index spaces.

Each symbol knows its type so we can still distinguish between them if needed.

Reply:
I think I'm swayed, I'll make it a single index space.

I'll update the wording of the Linking.md when I'm out of a meeting, and make it into a PR into the main repo.